source/remoteClient/client.py

```diff
@@ -0,0 +1,483 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2015-2025 NV Access Limited, Christopher Toth, Tyler Spivey, Babbage B.V., David Sexton and others.
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+import threading
+from typing import Callable, Optional, Set, Tuple
+
+import api
+import braille
+import core
+import gui
+import inputCore
+import ui
+import wx
+from config import isInstalledCopy
+from keyboardHandler import KeyboardInputGesture
+from logHandler import log
+from utils.alwaysCallAfter import alwaysCallAfter
+from utils.security import isRunningOnSecureDesktop
+
+from . import configuration, cues, dialogs, serializer, server, url_handler
+from .connectionInfo import ConnectionInfo, ConnectionMode
+from .localMachine import LocalMachine
+from .menu import RemoteMenu
+from .protocol import RemoteMessageType, addressToHostPort
+from .secureDesktop import SecureDesktopHandler
+from .session import MasterSession, SlaveSession
+from .protocol import hostPortToAddress
+from .transport import RelayTransport
+
+# Type aliases
+KeyModifier = Tuple[int, bool]  # (vk_code, extended)
+Address = Tuple[str, int]  # (hostname, port)
+
+
+class RemoteClient:
+	localScripts: Set[Callable]
+	localMachine: LocalMachine
+	masterSession: Optional[MasterSession]
+	slaveSession: Optional[SlaveSession]
+	keyModifiers: Set[KeyModifier]
+	hostPendingModifiers: Set[KeyModifier]
+	connecting: bool
+	masterTransport: Optional[RelayTransport]
+	slaveTransport: Optional[RelayTransport]
+	localControlServer: Optional[server.LocalRelayServer]
+	sendingKeys: bool
+
+	def __init__(
+		self,
+	):
+		log.info("Initializing NVDA Remote client")
+		self.keyModifiers = set()
+		self.hostPendingModifiers = set()
+		self.localScripts = set()
+		self.localMachine = LocalMachine()
+		self.slaveSession = None
+		self.masterSession = None
+		self.menu: Optional[RemoteMenu] = None
+		if not isRunningOnSecureDesktop():
+			self.menu: Optional[RemoteMenu] = RemoteMenu(self)
+		self.connecting = False
+		self.URLHandlerWindow = url_handler.URLHandlerWindow(
+			callback=self.verifyAndConnect,
+		)
+		url_handler.register_url_handler()
+		self.masterTransport = None
+		self.slaveTransport = None
+		self.localControlServer = None
+		self.sendingKeys = False
+		self.sdHandler = SecureDesktopHandler()
+		if isRunningOnSecureDesktop():
+			connection = self.sdHandler.initializeSecureDesktop()
+			if connection:
+				self.connectAsSlave(connection)
+				self.slaveSession.transport.connectedEvent.wait(
+					self.sdHandler.SD_CONNECT_BLOCK_TIMEOUT,
+				)
+		core.postNvdaStartup.register(self.performAutoconnect)
+		inputCore.decide_handleRawKey.register(self.process_key_input)
```

Comment:
Looking at the documentation of this extension point, I noticed that it looks like it is wrong while the behavior is as expected. The docstring reads: `Handlers can decide whether the key should be processed by NVDA and/or passed to the OS.` However they can't do the latter. The key is either handled by NVDA (returning True) or completely blocked (returning False).